pyproject.toml

```diff
@@ -0,0 +1,55 @@
+[tool.poetry]
+name = "cyclic-boosting"
+version = "0.1.0"
+description = "Implementation of Cyclic Boosting machine learning algorithms"
+authors = ["Blue Yonder GmbH"]
+packages = [{include = "cyclic_boosting"}]
+readme = "README.md"
+classifiers = [
+   "Topic :: Scientific/Engineering :: Artificial Intelligence",
+   "Programming Language :: Python :: 3 :: Only",
+   "Programming Language :: Python", 
+   "License :: OSI Approved :: Eclipse Public License 2.0 (EPL-2.0)"
+]
+
+[tool.poetry.dependencies]
+python = "^3.8"
+six = "^1.16.0"
+decorator = "^5.1.1"
+numba = "^0.56.4"
+numba-scipy = "^0.3.1"
+numpy = ">=1.12.1"
+numexpr = ">=2.5.2"
+scikit-learn = ">=0.18.2"
+pandas = ">=0.20.3"
+matplotlib = ">=1.5.1"
+hypothesis = "^6.70.0"
+
+[tool.poetry.group.dev.dependencies]
+black = "^23.1.0"
+ruff = "^0.0.256"
+pytest = "^7.2.2"
+pre-commit = "^3.2.2"
+pytest-cov = "^4.0.0"
+
+[tool.poetry.group.doc.dependencies]
+pydata-sphinx-theme = "^0.13.1"
+myst-parser = "^1.0.0"
+Sphinx = "^6.1.3"
+
+[tool.poetry.group.jupyter.dependencies]
+jupyterlab = "^3.6.3"
+jupyter-black = "^0.3.3"
+
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
+
+
+[tool.black]
+line-length = 120
+target_version = ["py38"]
+
+[tool.ruff]
+line-length = 120
+ignore = ["E741", "E722"]
```

Comment:
@FelixWick : disabled using `l` as a variable name and unspecified `except`s. All other issues could be automatically fixed